addons/mass_mailing/models/mail_thread.py

```diff
@@ -50,3 +51,15 @@ def message_post_with_template(self, template_id, **kwargs):
             default_mass_mailing_id=False,
         )
         return super(MailThread, no_massmail).message_post_with_template(template_id, **kwargs)
+
+    @api.multi
+    def message_receive_bounce(self, email, partner, mail_id=None):
+        super(MailThread, self).message_receive_bounce(email, partner)
+        # Auto-blacklist mechanism
+        recipient_email = '%s <%s>' % (partner.display_name, email)
+        three_months = datetime.datetime.now() - datetime.timedelta(weeks=13)
+        stats = self.env['mail.mail.statistics'].search([('bounced', '>', fields.Datetime.to_string(three_months))])\
```

Comment:
I think searching on all possible bounced statistics will gradually become a performance issue :face_with_head_bandage: 

Shouldn't we search on statistics whose recipient ilike email or something like that instead ?